File: ruby/lib/sorpresa_porcasahotel.rb
module Civitas
    class SorpresaPorCasaHotel < Sorpresa
        public_class_method :new
        def initialize(texto, valor)
            super(texto)
            @valor=valor
        end
        
        def aplicarAJugador(actual, todos)
            if jugadorCorrecto(actual, todos)
                informe(actual, todos)
                todos[actual].modificarSaldo(@valor*todos[actual].cantidadCasasHoteles)
            end
        end
    end
end

File: java/src/civitas/SorpresaPorCasaHotel.java
package civitas;

import java.util.ArrayList;

public class SorpresaPorCasaHotel extends Sorpresa {
    private int valor;

    SorpresaPorCasaHotel(String texto, int valor) {
        super(texto);
        this.valor = valor;
    }
    
    public void aplicarAJugador(int actual, ArrayList<Jugador> todos) {
        if (jugadorCorrecto(actual, todos)){
            informe(actual,todos);
            todos.get(actual).modificarSaldo(valor * todos.get(actual).cantidadCasasHoteles());
        }
    }
}